transaction/core/src/tx.rs

```diff
@@ -531,4 +622,41 @@ mod tests {
         let recovered_tx: Tx = Tx::decode(&buf[..]).unwrap();
         assert_eq!(tx, recovered_tx);
     }
+
+    #[test]
+    fn test_tx_prefix_hash() {
+        let mut rng: StdRng = SeedableRng::from_seed([32u8; 32]);
+        let outputs = get_outputs(3, &mut rng);
+
+        let inputs = get_inputs(2, 4, &mut rng);
+        let prefix = TxPrefix::new(inputs.clone(), outputs.clone(), MINIMUM_FEE, 100);
+        let prefix_hash = prefix.hash();
+
+        // Verify that the prefix with the same contents produces the same hash
+        let prefix1 = TxPrefix::new(inputs.clone(), outputs.clone(), MINIMUM_FEE, 100);
+        let prefix_hash1 = prefix1.hash();
+        assert_eq!(prefix_hash, prefix_hash1);
+
+        // Now let's construct a different set of membership elements in the proof with everything else the same
```

Comment:
I would add another test where an existing proof is changed, to ensure that the hash is also different in that case. Depending on how thorough you need this to be, you might want to add a test that modifies each and every part of the proof to see that it yields a different hash.